code/datums/elements/embed.dm

```diff
@@ -0,0 +1,147 @@
+
+			return TRUE
+
+/datum/element/embed/proc/fall_out(obj/item/weapon, mob/living/carbon/human/victim)
+	var/obj/item/bodypart/L = weapon.embedded_in
+	var/datum/embedding_behavior/EB = weapon.embedding
+
+	if(weapon.is_embed_harmless())
+		victim.visible_message("<span class='danger'>[weapon] falls off of [victim.name]'s [L.name]!</span>","<span class='userdanger'>[weapon] falls off of your [L.name]!</span>")
+	else
+		var/damage = weapon.w_class * EB.fall_pain_mult
+		L.receive_damage(brute=(1-EB.pain_stam_pct) * damage, stamina=EB.pain_stam_pct * damage)
+		victim.visible_message("<span class='danger'>[weapon] falls out of [victim.name]'s [L.name]!</span>","<span class='userdanger'>[weapon] falls out of your [L.name]!</span>")
+
+	safe_remove(weapon, victim, L)
+	weapon.forceMove(victim.drop_location())
+
+/datum/element/embed/proc/rip_out(obj/item/weapon, mob/living/carbon/human/victim, obj/item/bodypart/L)
+	var/datum/embedding_behavior/EB = weapon.embedding
+	var/time_taken = EB.rip_time * weapon.w_class
+
+	victim.visible_message("<span class='warning'>[victim] attempts to remove [weapon] from [victim.p_their()] [L.name].</span>","<span class='notice'>You attempt to remove [weapon] from your [L.name]... (It will take [DisplayTimeText(time_taken)].)</span>")
+	if(do_after(victim, time_taken, needhand = 1, target = victim))
+		if(!weapon || !L || weapon.loc != victim || !(weapon in L.embedded_objects))
+			active_embeds -= weapon
+			if(!victim.has_embedded_objects())
+				UnregisterSignal(victim, COMSIG_MOVABLE_MOVED)
+			return
+
+		if(weapon.is_embed_harmless())
+			victim.visible_message("<span class='notice'>[victim] successfully unsticks [weapon] from [victim.p_their()] [L.name]!</span>", "<span class='notice'>You successfully unstick [weapon] from your [L.name].</span>")
+		else
+			var/damage = weapon.w_class * EB.rip_pain_mult
+			L.receive_damage(brute=(1-EB.pain_stam_pct) * damage, stamina=EB.pain_stam_pct * damage) //It hurts to rip it out, get surgery you dingus.
+			victim.emote("scream")
+			victim.visible_message("<span class='notice'>[victim] successfully rips [weapon] out of [victim.p_their()] [L.name]!</span>", "<span class='notice'>You successfully remove [weapon] from your [L.name].</span>")
+
+		safe_remove(weapon, victim, L)
+		victim.put_in_hands(weapon)
+
+// this proc merely removes the embedded item from the person and from processing, and is still called at the end of painful rip-outs despite the name
+/datum/element/embed/proc/safe_remove(obj/item/weapon, mob/living/carbon/human/victim, obj/item/bodypart/L)
+	L.embedded_objects -= weapon
+	weapon.embedded_in = null
+	weapon.forceMove(get_turf(victim))
+	active_embeds -= weapon
+
+	if(!victim.has_embedded_objects())
+		victim.clear_alert("embeddedobject")
+		UnregisterSignal(victim, COMSIG_MOVABLE_MOVED)
+		SEND_SIGNAL(victim, COMSIG_CLEAR_MOOD_EVENT, "embedded")
+
+/datum/element/embed/process()
+	for(var/obj/item/weapon in active_embeds)
```

Comment:
If you know that only weapons will be in the list you don't need a filtered loop